ngraph/core/src/op/einsum.cpp

```diff
@@ -0,0 +1,299 @@
+            is_ellipsis_met = true;
+
+            // make additional increment since ellipsis consists of three dots.
+            ch_idx += 2;
+        }
+        else if (std::isalpha(subscript[ch_idx]) == 0)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void op::v7::Einsum::parse_equation(const std::string& equation,
+                                    std::vector<std::string>& input_subscripts,
+                                    std::string& output_subscript)
+{
+    NGRAPH_OP_SCOPE(v7_Einsum_parse_equation);
+
+    // split equation to input subscripts and an output subscript
+    auto pos_output_delimeter = equation.find("->");
+    auto input_subscripts_str = equation.substr(0, pos_output_delimeter);
+
+    // split the input subscripts into a vector of input subscripts
+    bool is_ellipsis_met = false;
+    input_subscripts.clear();
+    std::istringstream input;
+    input.str(input_subscripts_str);
+    for (std::string input_subscript; std::getline(input, input_subscript, ',');)
+    {
+        bool local_is_ellipsis_met = false;
+        // check that input subscript contains only alphabetic letter or ellipsis
+        NGRAPH_CHECK(is_subscript_correct(input_subscript, local_is_ellipsis_met) == true,
+                     "Input subscript of Einsum equation must consist of either only "
+                     "alphabetic letters or alphabetic letters with one ellipsis.");
+
+        // mark that ellipsis is met at least in one input subscript
+        if (local_is_ellipsis_met == true)
+        {
+            is_ellipsis_met = true;
+        }
+        input_subscripts.push_back(input_subscript);
+    }
+
+    if (pos_output_delimeter == std::string::npos)
+    {
+        // recover output subscript
+        output_subscript = "";
+        for (auto const& input_subscript : input_subscripts)
+        {
+            for (auto const& label : input_subscript)
+            {
+                if (std::isalpha(label) && output_subscript.find(label) == std::string::npos)
+                {
+                    output_subscript += label;
+                }
+            }
+        }
+        std::sort(output_subscript.begin(), output_subscript.end());
+        if (is_ellipsis_met)
+        {
+            output_subscript = "..." + output_subscript;
+        }
+    }
+    else
+    {
+        output_subscript = equation.substr(pos_output_delimeter + 2);
+        bool local_is_ellipsis_met = false;
+
+        // check that the output subscript has the correct format
+        NGRAPH_CHECK(is_subscript_correct(output_subscript, local_is_ellipsis_met),
+                     "Output subscript of Einsum equation must consist of either only "
+                     "alphabetic letters or alphabetic letters with one ellipsis.");
+
+        // if the ellipsis is met in input subscripts, one ellipsis must be in the output subscript
+        NGRAPH_CHECK(is_ellipsis_met == false || local_is_ellipsis_met == true,
```

Comment:
Change to `(is_ellipsis_met ^ local_is_ellipsis_met) == false`. The logic behind is that both variables should be either `true` or `false`. And the XOR operation should be used in such a cases.

Reply:
Great catch! But I think it can be simplified as `is_ellipsis_met == output_is_ellipsis_met`

Reply:
Done and added a test.